src/lib.rs

```diff
@@ -37,13 +38,14 @@ pub trait OrderedMapIterator: Iterator<Item=(<Self as OrderedMapIterator>::Key,
         }
     }
 
-    /// Join an ordered iterator with the right ordered iterator. The
-    /// new iterator will return a key value pair for every key in
-    /// either iterator. If a key is present in both iterators they
-    /// will be returned together (two values). If a value is in the Right,
-    /// but not the left iterator it will be return without the value in the
-    /// left iterator. If the value is in the left iterator by not the right
-    /// that will be return without the value from the left iterator.
+    /// Joins an ordered iterator with the right ordered iterator.
```

Comment:
I was wondering about that myself, but wasn't sure. Let me double check before updating the doc.